### ISSUE TITLE
[UX] [Ray Dashboard] Download log files without waiting for them to load

### ISSUE THREAD

**Issue by shrekris-anyscale** [ray-project/ray#42346]
### What's the context?

Serve's replicas, http proxies, and controller are all long-lived actors. When their logs fill up, Ray automatically starts a new log file:

<details>
  <summary>Example log structure</summary>

  
<img width="542" alt="Screen Shot 2023-08-03 at 12 30 27 PM" src="https://github.com/anyscale/product/assets/92341594/534d1ac6-75b7-4f2d-b15e-54995961eb0a">

</details>

I can download these log files one-by-one by clicking on their link in the dashboard and clicking `Download`:

<details>
  <summary>Example log</summary>

<img width="724" alt="Screen Shot 2023-08-03 at 12 37 29 PM" src="https://github.com/anyscale/product/assets/92341594/55890156-238e-494d-a33a-8e615a5952d1">

</details>

However, this causes 2 problems:

1. I can't download multiple log files at once. Since Serve's actors are long-lived, I need to download multiple log files to see all the logs for one actor.
2. I have to wait for the log file to load before I can access the download button. This can take 10's of seconds for a single log file, even if the file is empty (such as the one in the example above).

### What's the friction? What would be the ideal UX?

Ideally, I would select all the log files I want to download and download them at once without waiting. For example, there could be checkboxes in the log directory that would let me select and download multiple logs without loading any of them.

### Is there a workaround?

Yes, I have to wait for the log files to load before I can download them.

**Comment by shrekris-anyscale** [comment]
From @rynewang:

+1 on this. My use case is that I am debugging memory issues in Ray, so I followed the Ray document https://docs.ray.io/en/latest/ray-observability/user-guides/debug-apps/debug-memory.html to use `memray` which writes to the worker node's log. I can see the log in the Dash, but it's a ~6GB binary so I can't open it in the log details page. I ended up manually edited the URL to download it but if we have a "download" button in the log viewer page that'd be great.

![image](https://github.com/anyscale/product/assets/56065503/f99af145-0949-459f-acf0-3b7eea3f2fef)
